atc/worker/container_spec.go

```diff
@@ -42,9 +43,6 @@ type ContainerSpec struct {
 	User string
 }
 
-// The below methods cause ContainerSpec to fulfill the
-// go.opentelemetry.io/otel/api/propagation.HTTPSupplier interface
```

Comment:
I felt it was better to use a compile-time check rather than a comment to express intent
https://github.com/concourse/concourse/blob/f0478ad034157cd34cee71cce4040ac7ce6611ad/atc/worker/container_spec.go#L82

but yeah, explaining *why* it has to satisfy the interface is a good idea